File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/helper/DialogHelper.java
package com.dbottillo.mtgsearchfree.helper;

import android.support.v4.app.DialogFragment;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;

public class DialogHelper {

    public static void open(AppCompatActivity activity, String tag, DialogFragment fragment) {
        FragmentTransaction ft = activity.getSupportFragmentManager().beginTransaction();
        Fragment prev = activity.getSupportFragmentManager().findFragmentByTag(tag);
        if (prev != null) {
            ft.remove(prev);
        }
        ft.addToBackStack(null);
        fragment.show(ft, tag);
    }
}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/helper/DialogHelper.kt
package com.dbottillo.mtgsearchfree.helper

import android.support.v4.app.DialogFragment
import android.support.v7.app.AppCompatActivity

class DialogHelper {

    companion object {
        fun open(activity: AppCompatActivity, tag: String, fragment: DialogFragment) {
            val ft = activity.supportFragmentManager.beginTransaction()
            val prev = activity.supportFragmentManager.findFragmentByTag(tag)
            if (prev != null) {
                ft.remove(prev)
            }
            ft.addToBackStack(null)
            fragment.show(ft, tag)
        }
    }
}

